File: testdata/bench/swift/BenchmarkTest00022.swift
import Vapor

func handler022(_ req: Request) throws -> Response {
    let search: String = try req.query.get(at: "q")
    let body = "<div class='results'><p>Results for: \(search)</p></div>"
    var headers = HTTPHeaders()
    headers.add(name: .contentType, value: "text/html")
    return Response(status: .ok, headers: headers, body: .init(string: body))
}

File: testdata/bench/kotlin/BenchmarkTest00022.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.http.*

// Vulnerable: XSS via string concatenation in HTML
suspend fun handler00022(call: ApplicationCall) {
    val query = call.request.queryParameters["q"]
    val html = "<html><body><h1>Search results for: " + query + "</h1></body></html>"
    call.respondText(html, ContentType.Text.Html)
}
